File: 02-print-hello-world-loop/js/hello_world_loop_02.js
#!/usr/bin/env node

const { Command } = require('commander');
const program = new Command();



function main() {
    var repeat = 100;

    // Opciones

    program
        .name('string-util')
        .description('Hello world loop')
        .version('1.0.0')
        .usage('[-h] [-v] [-r REPEAT]')
        .option('-r, --repeat <REPEAT>', 'N times repeat "Hello world". (Default=100)', parseInt)
        .parse(process.argv);

    const options = program.opts();

    if (options.repeat) {
        repeat=options.repeat
    }

    //Inicio tiempo
    const time = process.hrtime();
    const NS_PER_SEC = 1e9;
    const MS_PER_NS = 1e-6
    //const time = process.hrtime(); TODO otro método
    // bucle
    for (let i=1;i<=repeat; i++) {
        console.log('Hello World',i);
    }

    //Fin tiempo
    //   const end = process.hrtime(start); TODO otro método

    //Impresión tiempo tardado
    //   console.log(`Spent time ${((end[0])+(end[1])/1000000000).toFixed(3)}s. to print ${repeat} times.`) TODO otro método
    const diff = process.hrtime(time);
    console.log(`Spent time ${ (((diff[0] * NS_PER_SEC + diff[1])  * MS_PER_NS)/1000).toFixed(3) } s. to print ${repeat} times.`)
}

main ();

File: 02-print-hello-world-loop/py/hello_world_loop_02.py
#!/usr/bin/env python3

#----------------------------------------------------------------#
#  Hello world loop 02
#----------------------------------------------------------------#
version = "1.0.0"

#---------------------------------------#
#       Code
#---------------------------------------#
import time
import argparse

#-----------------------------------------------#
# Mide tiempo
#-----------------------------------------------#
def mide_tiempo(funcion):
    def funcion_medida(*args, **kwargs):
        inicio = time.time()
        c = funcion(*args, **kwargs)
        elapsed_time=time.time()-inicio
        print("Spent time: %.10f s. to print" % elapsed_time)
        return c
    return funcion_medida

#-----------------------------------------------#
# Hello loop con parámetro
#-----------------------------------------------#
@mide_tiempo
def hello_loop_t(repeat):
    repeat=int(repeat)
    for x in range(1,repeat+1):
        print(x ,'Hello World')

#-----------------------------------------------#
# Hello loop default
#-----------------------------------------------#
@mide_tiempo
def hello_loop_d():
    num=100
    for x in range(1,num+1):
        print(x ,'Hello World')



#-----------------------------------------------#
#   Main
#-----------------------------------------------#
if __name__ == '__main__':
    parser = argparse.ArgumentParser(
        description="Hello world loop"
    )
    parser.version = 'Hello_word02_1.0.0'

    parser.add_argument(
        '-r',
        '--repeat',
        help="N times repeat Hello world. (Default=100)"
    )

    parser.add_argument(
        '-v',
        '--version',
        action='version',
        help="Version del programa"
    )

    args = parser.parse_args()

    if not args.repeat:
        hello_loop_d()

    else:
        hello_loop_t(
            repeat=args.repeat,
        )
